facade/src/main/scala/reactST/reactTable/HTMLTableBuilder.scala

```diff
@@ -0,0 +1,274 @@
+package reactST.reactTable
+
+import japgolly.scalajs.react._
+import japgolly.scalajs.react.internal.JsUtil
+import japgolly.scalajs.react.raw.React.ComponentClassP
+import japgolly.scalajs.react.vdom.html_<^._
+import react.common.Css
+import react.virtuoso.Virtuoso
+import reactST.reactTable.anon.Data
+import reactST.reactTable.mod.ColumnInterfaceBasedOnValue._
+import reactST.reactTable.mod.{ ^ => _, _ }
+import reactST.reactTable.syntax._
+import reactST.std.Partial
+
+import scalajs.js
+import scalajs.js.|
+import scalajs.js.JSConverters._
+import TableMaker.LayoutMarker
+import reactTableStrings._
+
+object HTMLTableBuilder {
+
+  /**
+   * Create a table component based on the configured plugins.
+   *
+   * @param tableMaker The TableMaker providing the useTable hook.
+   * @param options The table options to use for the table.
+   * @param headerCellFn A function to use for creating the <th> elements. Simple ones are provided by the TableMaker object.
+   * @param tableClass An optional CSS class to apply to the table element
+   * @param rowClassFn An option function that takes the index and the rowData and creates a class for the row. Relying on index does not work well for sortable tables since it is the unsorted index.
+   * @param footer An optional <tfoot> element. (see note below)
+   *
+   * Notes:
+   * If optimization is required, parts of the options and the data
+   *   should be memoized and only change when necessary. See react-table
+   *   documentation for details.
+   * The headerCellFn is expected to return a <th> element. The function
+   *   should make use of the column instance "props" methods.
+   * The typescript definitions don't seem to include the react-table
+   *   support for footers. So, they aren't available here, either.
+   *   So, table footers are fully "Build It Yourself". This actually
+   *   has some advantages, because it is often useful for table footers
+   *   to span multiple columns and react-table does not support that.
+   *   At some point, something similar for an "extra" header row might
+   *   be useful since header groups have some issues.
+   */
+  // format: off
+  def buildComponent[D,
```

Comment:
Should we make the names of buildComponent and makeVirtualizedTable more consistent?

Reply:
Yes, definitely. Missed that. Let me make a new commit addressing this and other comments.